main/EventSystem.cpp

```diff
@@ -3170,12 +3170,13 @@ bool CEventSystem::processLuaCommand(lua_State *lua_state, const std::string &fi
 		if (aParam.size() > 3) {
 			sound = aParam[3];
 		}
-		if (aParam.size() > 4) {
-			if (aParam[4].find("midx_") != std::string::npos) {
-				extraData = aParam[4];
+		if (aParam.size() > 4 && !aParam[4].empty()) {
+			if (isInt(aParam[4]) && aParam.size() > 5 && !aParam[5].empty() && aParam[5] == "fcm") {
+					// Only FCM notification Subsystem handles extraData "|Device=<deviceIdx>"
+					extraData = "|Device=" + aParam[4];
```

Comment:
So, as for previous comment, I think you should not start with a pipe...

Reply:
It isn't as nice as it could be indeed.
Current 'problem' is that the presence of a pipe separator is the identifying characteristic that 1 or more key/value-pairs are available in the extraData. If there is only 1 key/value-pair (like in this case), there would be no separator and how than determine the difference between key/value-pair(s) and _normal_ extraData :) ?